pkg/controllers/nodepool/hash/controller.go

```diff
@@ -69,3 +89,30 @@ func (c *Controller) Builder(_ context.Context, m manager.Manager) operatorcontr
 		WithOptions(controller.Options{MaxConcurrentReconciles: 10}),
 	)
 }
+
+func (c *Controller) updateNodeClaimHash(ctx context.Context, np *v1beta1.NodePool, nphash string) error {
+	ncList := &v1beta1.NodeClaimList{}
+	if err := c.kubeClient.List(ctx, ncList); err != nil {
+		return client.IgnoreNotFound(err)
+	}
+
+	for i := range ncList.Items {
```

Comment:
```suggestion
	for i, nc := range ncList.Items {
```

Is there something preventing us from assigning the NodeClaim to a variable in the range loop here?

Reply:
Removed for this design.